semanticdb/javac/src/main/scala/scala/meta/internal/semanticdb/javac/semantics/Elements.scala

```diff
@@ -0,0 +1,243 @@
+package scala.meta.internal.semanticdb.javac.semantics
+
+import javax.lang.model.`type`.TypeKind
+import javax.lang.model.element._
+import scala.collection.mutable
+import scala.meta.internal.{semanticdb => s}
+import scala.meta.internal.semanticdb.SymbolInformation.{Kind => k}
+import scala.meta.internal.semanticdb.Accessibility.{Tag => a}
+import scala.meta.internal.semanticdb.SymbolInformation.{Property => p}
+import scala.collection.JavaConverters._
+import scala.meta.internal.semanticdb.Scala.{Descriptor => d, _}
+
+trait Elements {
+
+  implicit class ElementOps(elem: Element) {
+
+    def enclosingPackage: PackageElement = elem.getEnclosingElement match {
+      case enclosing: PackageElement => enclosing
+      case enclosing => enclosing.enclosingPackage
+    }
+
+    def enclosedElements: Seq[Element] = elem.getEnclosedElements.asScala
+
+    def annotations: Seq[s.Annotation] = elem match {
+      case elem: ExecutableElement if elem.getModifiers.contains(Modifier.STRICTFP) =>
+        Seq(
+          s.Annotation(
+            tpe = s.TypeRef(symbol = "scala/annotation/strictfp#")
+          ))
+      case _ => Seq()
+    }
+
+    def isStatic(elem: ExecutableElement): Boolean = elem.getModifiers.contains(Modifier.STATIC)
+
+    def owner: String = elem.getEnclosingElement.sym
+
+    def sym: String = elem match {
+      case elem: PackageElement =>
+        val qualName = elem.getQualifiedName.toString
+        if (qualName == "") Symbols.EmptyPackage
+        else qualName.replace('.', '/') + "/"
+      case elem: TypeElement =>
+        Symbols.Global(owner, d.Type(name))
+      case elem: ExecutableElement =>
+        val owner = elem.getEnclosingElement
+        val disambig = {
+          val siblings = owner.enclosedElements
+          val siblingMethods = siblings.collect {
+            case sibling: ExecutableElement if sibling.name == name => sibling
+          }
+          val (instance, static) = siblingMethods.partition(method => !isStatic(method))
+          val methodPlace = (instance ++ static).indexOf(elem)
+          if (methodPlace == 0) "()"
+          else s"(+$methodPlace)"
+        }
+        Symbols.Global(owner.sym, d.Method(name, disambig))
+      case elem: VariableElement if elem.getKind == ElementKind.PARAMETER =>
+        Symbols.Global(owner, d.Parameter(name))
+      case elem: VariableElement =>
+        Symbols.Global(owner, d.Term(name))
+      case elem: TypeParameterElement =>
+        Symbols.Global(owner, d.TypeParameter(name))
+    }
+
+    def name: String = elem match {
+      case elem: PackageElement =>
+        if (elem.isUnnamed) "_empty_"
+        else {
+          val fullName = elem.getSimpleName.toString
+          fullName.substring(fullName.lastIndexOf('.') + 1)
+        }
+      case elem => elem.getSimpleName.toString
```

Comment:
I guess we're guaranteed constructors have `<init>` names?

Reply:
Yes. From the javac api:
```
For a constructor, the name {@code "<init>"} is
returned, for a static initializer, the name {@code "<clinit>"}
is returned, and for an anonymous class or instance
initializer, an empty name is returned.
```

Reply:
👍